### PR TITLE
[dvsim] Fix regression publish result link with --remote switch

### PR DESCRIPTION
**Pull request by cindychip**
If we use `--remote` switch, the result summary path is not correct.
This is because to remove the root directory and get `rel_path` in
FlowCfg, we used
```
        if self.rel_path == "":
            self.rel_path = os.path.dirname(self.flow_cfg_file).replace(
                self.proj_root + '/', '')
```
For this one, if remote switch is on, the self.proj_root is remote
repo_top. However, `self.flow_cfg_file` still uses the local abs path,
so the rel_path ended up still include the absolute path.
This PR fixed it by using the remote `cfg_path` if --remote is turned
on.

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

File: util/dvsim/dvsim.py [hunk 1]
```diff
@@ -601,9 +601,13 @@ def main():
...
     args.scratch_root = resolve_scratch_root(args.scratch_root)
     args.branch = resolve_branch(args.branch)
-    args.cfg = os.path.abspath(args.cfg)
     proj_root = resolve_proj_root(args)
     log.info("[proj_root]: %s", proj_root)
+    if args.remote:
+        cfg_path = os.getcwd().replace(get_proj_root() + "/", "")
+        args.cfg = os.path.join(proj_root, cfg_path, args.cfg)
+    else:
+        args.cfg = os.path.abspath(args.cfg)
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:610]: Something like this is probably a bit more clear. 
```suggestion
    # Have resolve_proj_root return both, the src and dest
    proj_root_src, proj_root_dest = resolve_proj_root(args)

    args.cfg = os.path.abspath(args.cfg)
    if args.remote:
        cfg_path = args.cfg.replace(proj_root_src + "/", "")
        args.cfg = os.path.join(proj_root_dest, cfg_path)
```

Comment (cindychip) [reply util/dvsim/dvsim.py]: Thanks Sri for the suggestion. Could you take another look and see if the updated change meets your expectation?

File: util/dvsim/dvsim.py [hunk 2]
```diff
@@ -601,10 +604,15 @@ def main():
 
     args.scratch_root = resolve_scratch_root(args.scratch_root)
     args.branch = resolve_branch(args.branch)
-    args.cfg = os.path.abspath(args.cfg)
-    proj_root = resolve_proj_root(args)
+    proj_root_src, proj_root_dest = resolve_proj_root(args)
+    proj_root = proj_root_dest
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:608]: I think the extra var is not required. Either keep `proj_root` or `proj_root_dest`, and update refs everywhere else.

Comment (cindychip) [reply util/dvsim/dvsim.py:608]: Sounds good! Thanks Sri, will update that!

**Review by sriyerg — APPROVED** [review]

**Pull request merged**